src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs

```diff
@@ -298,6 +314,11 @@ private IEnumerable<string> GetApplicationNames()
             throw new Exception("An application name must be provided");
         }
 
+        private bool ApplicationNameIsBlankOrDefault() => (_localConfiguration.application.name.Count == 0) ||
+            ((_localConfiguration.application.name.Count == 1) && (_localConfiguration.application.name[0] == DefaultAppNameInConfigFile));
+
+        private string GetLambdaFunctionName() => _environment.GetEnvironmentVariable("AWS_LAMBDA_FUNCTION_NAME");
```

Comment:
Could we push this down to config as a new property? `BootstrapConfig` already reads the value when checking for serverless mode; it could expose the value as a property that could be propagated up. We reference this same environment variable in the Lambda wrapper - it could be refactored to use the config property.

Reply:
Oh you mean `AWS_LAMBDA_FUNCTION_NAME`? Yeah that makes sense. Let me give it a try

Reply:
How does that look?

Reply:
I was thinking it would be a public property and that the reference to the environment variable in the lambda wrapper would be updated to use the property also.

Reply:
Ah I see what you mean now. Moving the env var calls out of the wrapper also sounds like a good idea.